main.js

```diff
@@ -15,45 +17,72 @@ lockButton.addEventListener('click', function(event) {
     }
 });
 
-newPaletteButtonSection.addEventListener('click', function(event) {
-    if(event.target.classList.contains('button-box-l') || event.target.id === 'new-palette' || event.target.classList.contains('button-box-r')) {
+buttonSection.addEventListener('click', function(event) {
+    if(event.target.classList.contains('new-palette')) {
         getNewHexes();
     }
 });
 
 function getNewHexes() {
-    currentHexes = [];
+    var oldHexes = currentPalette
+    currentPalette = [];
     var newColor;
     for(i = 0; i < mainColorBoxes.length; i++) {
         var thisColorBoxLock = mainColorBoxes[i].firstElementChild.firstElementChild
         if(thisColorBoxLock.classList.contains('unlocked')) {
             newColor = getRandomHex().toUpperCase();
             mainColorBoxes[i].firstElementChild.style.backgroundColor = `#${newColor}`;
             mainColorBoxes[i].lastElementChild.innerText = `#${newColor}`;
+            currentPalette.push(newColor);
+        } else { 
+            currentPalette.push(oldHexes[i])
         }
     }
 }
 
 function lockToggle(event) {
     if (event.getAttribute('src') === './assets/unlocked.png') {
         event.src = './assets/locked.png';
-        toggleLockClass(event);
     } else {
-        event.src = './assets/unlocked.png';
-        toggleLockClass(event);
-    }
-}
-
-function toggleLockClass(element) {
-    if(element.classList.contains('locked')) {
-        element.classList.remove('locked');
-        element.classList.add('unlocked');
-    } else {
-        element.classList.add('locked');
-        element.classList.remove('unlocked');
+        event.src = './assets/unlocked.png'; 
     }
+    event.classList.toggle('unlocked')
 }
 
 function getRandomHex() {
     return (Math.floor(Math.random() * 16777216).toString(16).padStart(6, 0));
 }
+
+buttonSection.addEventListener('click', function(event) {
+    if (event.target.classList.contains('save-palette')) {
+        savePalette();
+    }
+})
+
+function savePalette() {
+    if (!savedPalettes.includes(currentPalette)) {
+        savedPalettes.push(currentPalette);
+    }
```

Comment:
NB:  it looks like this is always returning true - we may need to look into `.every()` to check the actual contents of each array for the uniqueness extension ticket coming up.

<img width="1383" alt="Screenshot 2023-04-13 at 5 35 48 PM" src="https://user-images.githubusercontent.com/32660020/231889193-9930097b-9cd3-4cf3-ad85-214eeee9b2c9.png">